core/tests/protractor-browserstack.conf.js

```diff
@@ -0,0 +1,219 @@
+  // ----- How to setup Selenium -----
+  //
+  // There are three ways to specify how to use Selenium. Specify one of the
+  // following:
+  //
+  // 1. seleniumServerJar - to start Selenium Standalone locally.
+  // 2. seleniumAddress - to connect to a Selenium server which is already
+  //    running.
+  // 3. sauceUser/sauceKey - to use remote Selenium servers via SauceLabs.
+  //
+  // If the chromeOnly option is specified, no Selenium server will be started,
+  // and chromeDriver will be used directly (from the location specified in
+  // chromeDriver)
+
+  // The location of the selenium standalone server .jar file, relative
+  // to the location of this config. If no other method of starting selenium
+  // is found, this will default to
+  // node_modules/protractor/selenium/selenium-server...
+  seleniumServerJar: null,
+  // The port to start the selenium server on, or null if the server should
+  // find its own unused port.
+  seleniumPort: null,
+  // Chromedriver location is used to help the selenium standalone server
+  // find chromedriver. This will be passed to the selenium jar as
+  // the system property webdriver.chrome.driver. If null, selenium will
+  // attempt to find chromedriver using PATH.
+  chromeDriver: './selenium/chromedriver',
+  // If true, only chromedriver will be started, not a standalone selenium.
+  // Tests for browsers other than chrome will not run.
+  chromeOnly: false,
+  // Additional command line options to pass to selenium. For example,
+  // if you need to change the browser timeout, use
+  // seleniumArgs: ['-browserTimeout=60'],
+  seleniumArgs: [],
+
+  // If sauceUser and sauceKey are specified, seleniumServerJar will be ignored.
+  // The tests will be run remotely using SauceLabs.
+  sauceUser: null,
+  sauceKey: null,
+
+  // The address of a running selenium server. If specified, Protractor will
+  // connect to an already running instance of selenium. This usually looks like
+  // seleniumAddress: 'http://localhost:4444/wd/hub'
+  seleniumAddress: 'http://hub-cloud.browserstack.com/wd/hub',
+
+  // The timeout for each script run on the browser. This should be longer
+  // than the maximum time your application needs to stabilize between tasks.
+  // (Note that the hint tooltip has a 60-second timeout.)
+  allScriptsTimeout: 180000,
+
+  // ----- What tests to run -----
+  //
+  // When run without a command line parameter, all suites will run. If run
+  // with --suite=smoke, only the patterns matched by that suite will run.
+  suites: {
+    // When running the tests on browserstack, all suites should be added
+    // in 'full' individually. This is to parallelize the suites on
+    // browserstack.
+    full: [
+      'protractor/accessibility.js',
+      'protractor/learnerFlow.js',
+      'protractor/libraryFlow.js',
+      'protractor/loginFlow.js',
+      'protractor_mobile/navigation.js',
+      'protractor_mobile/playerFlow.js',
+      'protractor/ratings.js',
```

Comment:
So ... this has disappeared from the lists below but it is still here. I know it's an easy fix to delete this line (or reinstate the thing below) but I want to ask how we're going to make sure things stay consistent in the first place? I.e. can we do some kind of lint check for this file perhaps, if protractor/* and protractor_mobile/* is not an option? The fact that this has happened twice already shows quite clearly that such a check is needed.

Reply:
Added a check which performs the following:
1. Matches the number of suites listed in full to the actual number of suites.
2. Matches the spec files listed in full to the actual spec files.